source/textInfos/offsets.py

```diff
@@ -363,7 +362,7 @@ def _getCharacterOffsets(self, offset):
 		return (offset, offset + 1)
 
 	def _getWordOffsets(self,offset):
-		if self.encoding not in (textUtils.WCHAR_ENCODING, None, "utf_32_le", locale.getlocale()[1]):
+		if self.encoding not in (textUtils.WCHAR_ENCODING, None, "utf_32_le", textUtils.USERANSICODEPAGE):

```

Comment:
This was the line where this failed on I believe. The error could actually have been avoided by doing the following:

```suggestion
		if self.encoding != textUtils.WCHAR_ENCODING and self.encoding != None and self.encoding != "utf_32_le" and self.encoding != textUtils.USERANSICODEPAGE):

```

While this seems overkill, it ensures that it only evaluates the necessary part of the if statement. When applied to the old case it would never have reached the part that executed locale.getlocale(). It did before, as we were constructing the tuple first and then evaluated against every entry in the tuple.

Reply:
ugh, let me correct that. it should probably have been something like:

```suggestion
		if not (self.encoding == textUtils.WCHAR_ENCODING or self.encoding == None or self.encoding == "utf_32_le" or self.encoding == textUtils.USERANSICODEPAGE):

```